From fd13a4d304da4233cb954329bf287ec9dfbb7367 Mon Sep 17 00:00:00 2001
From: Jon Mason <jon.mason@arm.com>
Date: Mon, 4 Dec 2023 10:20:21 -0500
Subject: [PATCH] bl31_runtime: revert usage of plat_ic_has_interrupt_type

There is a regression caused by commit
1f6bb41dd951714b47bf07bb9a332346ca261033 for the trusted services tests.
This is due to the fact that the referenced commit changes the behavior
from checking for both INTR_TYPE_EL3 and INTR_TYPE_S_EL1, to referencing
an existing function that #if for _either_ INTR_TYPE_EL3 or
INTR_TYPE_S_EL1 (depending on the value of GICV2_G0_FOR_EL3).  To work
around this issue, revert the check back to its original form.

Signed-off-by: Jon Mason <jon.mason@arm.com>
Upstream-Status: Pending
---
 bl31/interrupt_mgmt.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bl31/interrupt_mgmt.c b/bl31/interrupt_mgmt.c
index 68c7f10add21..8e888b676b35 100644
--- a/bl31/interrupt_mgmt.c
+++ b/bl31/interrupt_mgmt.c
@@ -47,9 +47,9 @@ static intr_type_desc_t intr_type_descs[MAX_INTR_TYPES];
  ******************************************************************************/
 static int32_t validate_interrupt_type(uint32_t type)
 {
-	if (plat_ic_has_interrupt_type(type)) {
+	if ((type == INTR_TYPE_S_EL1) || (type == INTR_TYPE_NS) ||
+	    (type == INTR_TYPE_EL3))
 		return 0;
-	}
 
 	return -EINVAL;
 }
-- 
2.30.2

